cli/src/main.rs

```diff
@@ -230,6 +230,66 @@ fn build_app<'a, 'b>() -> App<'a, 'b> {
                                     .help("Filters trades that are a part of a particular offer_id"),
                             )
                     ),
+                )
+                .subcommand(
+                    SubCommand::with_name("aggregations")
+                        .about("Fetch aggregate statistics over a specified time range.")
+                        .arg(
+                            Arg::with_name("start_time")
```

Comment:
This is an interesting idea.  Are you suggesting that we update the fields in the resource and the endpoint such that they are DateTime objects, or are you saying that the CLI should take ISO8601 as an input because it is easier for a human to write, but keep milliseconds everywhere else?  I'm inclined to say that if we are going down this path we should make everything a DateTime.

The only potential drawback is that now the resolution and the dates are in different formats so there may be some extra calculations a user is responsible to make.  This may be offset by the fact that no calculations are required to generate the dates.

These arguments are optional in the CLI because they are optional in the endpoint.  Same point about wanting to update the endpoint if we change things here.

Reply:
1. The resolution is a "duration" so it's definitely not the same data type, but it _could_ be a duration struct instead. It would make working with this end point slightly easier if our interface used the chrono datetime. As for a the duration in the cli. It might make sense to create an enum of common durations. The endpoint actually says it only supports a few resolutions but it actually accepts many, at least when I was testing it.

> segment duration as millis since epoch. Supported values are 5 minutes (300000), 15 minutes (900000), 1 hour (3600000), 1 day (86400000) and 1 week (604800000).

2. The start_time/end_time aren't actually optional, they just initialize to nonsense, but if you omit either one the pagination will be broken. We should probably update the endpoint to require passing in some sort of "span" or "range" in order to create the endpoint struct.